Sprint 111 blog post (#342)
openSUSE/obs-landing: _posts/development/2022-02-03-scm-integration-report-improvements.md

---
layout: post
title: UI and Reporting Improvements for the Continuous Integration with OBS and GitHub/GitLab
category: development
---

Today, we have some improvements around the continuous integration we unveiled in [one of our previous blog posts](https://openbuildservice.org/2021/05/31/scm-integration/). We expanded the workflow run interface, added a new type of step to configure repository flags, and introduced a breaking change to the `configure_repositories` step.

As you know, this is a beta feature. So, do not forget this feature is under [the beta program](/2018/10/04/the-beta-program/). Join!

{% include partials/_series-of-posts-about-scm-integration.md %}

## Support multiple repository paths for the configure_repositories step

> You already know about the `configure_repositories` step. Until now a single path element is defined per repository configured, but we realized for some projects more than one path has to be defined. So, we have made improvements for the support of multiple path elements. The change has been documented and you can check it out [here](https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.scm_ci_workflow_integration.html#sec.obs.obs_scm_ci_workflow_integration.obs_workflows.steps.configure_repositories_architectures_for_a_project).

Reviewer: One last thing :innocent: 

In the introduction, you mentioned that there is a breaking change. However, in the body it's not clear which is the breaking one.

Probably a little change like this is enough:

 ```suggestion
You already know about the `configure_repositories` step. Until now a single path element is defined per repository configured, but we realized for some projects more than one path has to be defined. So, we have made improvements for the support of multiple path elements. The breaking change has been documented and you can check it out [here](https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.scm_ci_workflow_integration.html#sec.obs.obs_scm_ci_workflow_integration.obs_workflows.steps.configure_repositories_architectures_for_a_project).
```

Reviewer: You can even have something more evident like:

:warning: Breaking change :warning: 
Check the documentation [here](https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.scm_ci_workflow_integration.html#sec.obs.obs_scm_ci_workflow_integration.obs_workflows.steps.configure_repositories_architectures_for_a_project) and adapt your _configure_repositories_ step in your `.obs/workflows.yml`

---
layout: post
title: UI and Reporting Improvements for the Continuous Integration with OBS and GitHub/GitLab
category: development
---

> Today, we have some improvements around UI and reporting and user feedback for the continuous integration we unveiled in our [previous blog post](https://openbuildservice.org/2021/05/31/scm-integration/).

Reviewer 2: ```suggestion
Today, we have some improvements around the continuous integration we unveiled in our [previous blog post](https://openbuildservice.org/2021/05/31/scm-integration/). We expanded the workflow run interface, added a new type of step to configure repository flags and introduced a breaking change to the configure_repositories step.
```

> Today, we have some improvements around UI and reporting and user feedback for the continuous integration we unveiled in ours [previous blog post](https://openbuildservice.org/2021/05/31/scm-integration/).

Reviewer 2: ```suggestion
Today, we have some improvements around UI and reporting and user feedback for the continuous integration we unveiled in our [previous blog post](https://openbuildservice.org/2021/05/31/scm-integration/).
```

As you know, this is a beta feature. So, do not forget this feature is under the [the beta program](/2018/10/04/the-beta-program/). Join!

{% include partials/_series-of-posts-about-scm-integration.md %}

> ## WorkflowRuns artifacts

Reviewer 2: ```suggestion
## Artifact of Workflow Runs
```

> A single webhook can generate different artifacts on OBS. Artifacts are the projects or packages that are involved in the step definition given in `workflows.yml`.

Reviewer 2: ```suggestion
A single webhook on your SCM can trigger different workflows, given in `workflows.yml`, to run. Each Workflow Run will most likely generate some changes inside OBS project or packages. Those changes are what we call Workflow Run Artifacts. ```

Reviewer 3: In addition to @hennevogel's proposal, we should refer to `.obs/workflows.yml` instead of `workflows.yml`.

> For example, repositories and architectures if the step is `configure_repositories` or projects and packages in steps like `link_package` etc.

Reviewer 2: ```suggestion
For example, repositories created if the step is `configure_repositories` or projects and packages created in steps like `link_package` etc.
```

> After each workflow run, we now display all the artifacts involved so you can better understand what was going on.

Reviewer 2: ```suggestion
After each workflow run, we now display all the artifacts involved so you can better understand and debug what was going on.
```

- Push event artifacts.
<figure>
  <img src="/images/posts/sprint_111_artifacts.png" alt="Workflow run artifacts" />
  <figcaption>Workflow run artifacts</figcaption>
</figure>

## The set_flags step

> Projects and packages can have different flags based on repositories and architectures. With `set_flags` you have the opportunity to manage flags for your

Reviewer 2: I think you should explain a bit what this is about. Steal from the user docu 😃

projects and packages.

Here is a simple example of how you can use it:
```
workflow:
  steps:
    - set_flags:
        flags:
          - type: build
            status: enable
            project: home:Admin
          - type: publish
            status: disable
            project: home:Admin
```

> Please refer to the [documentation](https://github.com/openSUSE/open-build-service/wiki/Better-SCM-CI-Integration#set-flags) to learn more.

Reviewer 2: This is the developer documentation, the user documentation is in the books.

## Support multiple repository paths for the configure_repositories step
<!-- TODO -->

> ## Improvement for WorkflowRuns UI

Reviewer 2: ```suggestion
## Improvement for Workflow Runs User Interface
```

> WorkflowRuns is one of the most important places to debug your workflows. It helps you to fix the problems like the wrong configurations in `workflows.yml`.

Reviewer 2: ```suggestion
The Workflow Runs page is the most important place to debug your workflows. It helps you to fix the problems like the wrong configurations in `workflows.yml`.
```

> To make easier to debug and troubleshoot broken SCM integrations we have updated the `WorkflowRuns UI`.

Reviewer 2: ```suggestion
So to make it easier to debug and troubleshoot broken SCM integrations we have added some more feature to the Workflow Runs UI.
```

> ## Artifact of Workflow Runs

Reviewer 4: ```suggestion
## Artifacts of Workflow Runs
```

Reviewer 4: or
```suggestion
## Workflow Runs Artifacts
```

> ## Artifact of Workflow Runs

Reviewer: Well, Dani's suggestion sounds better: "Workflow Runs Artifacts"

 https://github.com/openSUSE/obs-landing/pull/342#discussion_r800494214

You will get details like:

> - A new show view for individual workflows has been added to help you share WorkflowRuns across your team.

Reviewer 2: ```suggestion
- A new show view for individual workflows has been added to help you share Workflow Runs across your team.
```

<figure>
  <img src="/images/posts/sprint_111_workflow_runs_show.png" alt="Workflow runs show view" />
  <figcaption>Workflow runs show view</figcaption>
</figure>

> - New filter options to help filter WorkflowRuns based on `status` and `event type`

Reviewer 2: ```suggestion
- New filter options to help filter Workflow Runs based on `status` and `event type`
```

> So to make it easier to debug and troubleshoot broken SCM integrations we have added some more features to the Workflow Runs UI.
> 
> You will get details like:
> 
> - The details about each workflow run are displayed on a new page.

Reviewer: ```suggestion
So to make it easier to debug and troubleshoot broken SCM integrations we have added some more features to the Workflow Runs UI:

- The details about each workflow run are displayed on a new page.
```

Here I suggest finishing the first paragraph with `:` and removing "You will get details like:".

>   <img src="/images/posts/sprint_111_workflow_runs_show.png" alt="Workflow runs show view" />

Reviewer: Sorry, I didn't realize before... this screenshot doesn't contain the Artifacts tab.
You can use this one instead:
https://openbuildservice.org/help/manuals/obs-user-guide/images/workflow_runs_request.png

> You will get details like:
> 
> - A new show view for individual workflows has been added to help you share Workflow Runs across your team.

Reviewer: The show view is still not shareable with someone else, so I would change the text a bit:

```suggestion
- The details about each workflow run are displayed on a new page.
- Together with the Request and Response, we added the new tab Artifacts.
```

> Example of artifacts for a push event:

Reviewer 4: ```suggestion
```

Reviewer 4: I would remove this line as suggested, because it's redundant with the caption of the image below...

Author: It's removed.

<figure>
  <img src="/images/posts/sprint_111_artifacts.png" alt="Workflow run artifacts" />

>   <figcaption>Workflow run artifacts</figcaption>

Reviewer 4: ```suggestion
  <figcaption>Workflow run artifacts for a push event</figcaption>
```
